litestar/handlers/base.py

```diff
@@ -511,7 +511,7 @@ def resolve_return_dto(self) -> type[AbstractDTO] | None:
 
         return self._resolved_return_dto
 
-    async def authorize_connection(self, connection: ASGIConnection) -> None:
+    async def authorize_connection(self, connection: ASGIConnection | Request | WebSocket) -> None:
```

Comment:
This doesn't make much sense IMO. These are all subclasses of `ASGIConnection`, and it's not guaranteed that it will be these concrete classes that are passed in, e.g. in the case of a custom `request_class`. It would also be considered a breaking change, since users sublclassing this would get type errors because of an incompatible override